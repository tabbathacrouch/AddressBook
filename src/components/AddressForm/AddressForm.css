.form {
  margin-top: 1.5rem;
  margin-left: 1.5rem;
  margin-right: 2.5rem;
  display: flex;
  flex-direction: column;
  font-size: 1rem;
}

.error {
  color: red;
  margin-left: 1rem;
}

.field {
  width: 100%;
  margin: 0.5rem;
  padding: 0.5rem;
  border-style: solid;
  border-width: 1px;
  border-radius: 0.25rem;
  box-shadow: 2px 2px 4px 2px rgba( 0, 0, 0, 0.2);
  font-size: 1rem;
}

.field:hover {
  background-color: #F4F4F4;
}

.city {
  width: 55%;
  margin-right: 2em;
}

.stateName {
  width: 15%;
}

.zip {
  margin-left: 2em;
}

.fieldRow {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.btn-row {
  width: 100%;
  display: flex;
  justify-content: center;
}

.submit-btn {
  border-style: solid;
  border-width: 1px;
  border-radius: 0.25rem;
  box-shadow: 2px 2px 4px 2px rgba( 0, 0, 0, 0.2);
  width: 25%;
  padding: 0.5rem;
  margin: 0.5rem;
  font-size: 1rem;
}

.submit-btn:hover {
  background-color: #333333;
  color: white;
}

.search-form {
  display: flex;
  justify-content: center;
}