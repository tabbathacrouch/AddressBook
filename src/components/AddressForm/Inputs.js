import React from "react";
import { Field, ErrorMessage } from "formik";

export const NameInput = () => (
  <div>
    <Field
      data-testid="name"
      name="name"
      type="text"
      placeholder="Name"
      className="field"
    />
    <ErrorMessage name="name">
      {(errorMessage) => (
        <div className="error" data-testid="nameError">
          {errorMessage}
        </div>
      )}
    </ErrorMessage>
  </div>
);

export const StreetInput = () => (
  <div>
    <Field
      data-testid="street"
      name="street"
      type="text"
      placeholder="Street"
      className="field"
    />
    <ErrorMessage name="street">
      {(errorMessage) => (
        <div className="error" data-testid="streetError">
          {errorMessage}
        </div>
      )}
    </ErrorMessage>
  </div>
);

export const CityInput = () => (
  <div className="city">
    <Field data-testid="city" name="city" type="text" placeholder="City" className="field" />
    <ErrorMessage name="city">
      {(errorMessage) => <div className="error" data-testid="cityError">{errorMessage}</div>}
    </ErrorMessage>
  </div>
);

export const StateInput = () => (
  <div className="stateName">
    <Field data-testid="stateName" name="stateName" type="text" placeholder="State" className="field" />
    <ErrorMessage name="stateName">
      {(errorMessage) => <div className="error" data-testid="stateNameError">{errorMessage}</div>}
    </ErrorMessage>
  </div>
);

export const ZipInput = () => (
  <div className="zip">
    <Field data-testid="zip" name="zip" type="text" placeholder="Zip" className="field" />
    <ErrorMessage name="zip">
      {(errorMessage) => <div className="error" data-testid="zipError">{errorMessage}</div>}
    </ErrorMessage>
  </div>
);
