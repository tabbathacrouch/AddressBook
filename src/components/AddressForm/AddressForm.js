import React from "react";
import { Formik, Form } from "formik";
import * as Yup from "yup";
import "./AddressForm.css";
import {
  NameInput,
  StreetInput,
  CityInput,
  StateInput,
  ZipInput,
} from "./Inputs";


export const addressFormValidator = Yup.object({
  name: Yup.string()
    .max(30, "Must be 30 characters or less")
    .matches(
      /^[aA-zZ\s]+$/,
      "Numbers and special characters are not allowed"
    )
    .required("Required"),
  street: Yup.string()
    .max(40, "Must be 40 characters or less")
    .required("Required"),
  city: Yup.string()
    .max(50, "Must be 50 characters or less")
    .matches(
      /^[aA-zZ\s]+$/,
      "Numbers and special characters are not allowed"
    )
    .required("Required"),
  stateName: Yup.string()
    .max(15, "Must be 15 characters or less")
    .matches(/^[aA-zZ\s]+$/, "Invalid State name")
    .required("Required"),
  zip: Yup.string()
    .matches(
      /^(?=(\D*\d){5}\D*$)(?=[^ ]* ?[^ ]*$)[\d ]*$/,
      "Must contain 5 digits"
    )
    .required("Required"),
});

const AddressForm = ({ addContact }) => {
  return (
    <Formik
      initialValues={{
        id: 0,
        name: "",
        street: "",
        city: "",
        stateName: "",
        zip: "",
      }}
      validationSchema={addressFormValidator}
      onSubmit={(contact, { resetForm }) => {
        addContact(contact);
        const prev = contact.id;
        resetForm({
          values: {
            id: prev + 1,
            name: "",
            street: "",
            city: "",
            stateName: "",
            zip: "",
          },
        });
      }}
    >
      <Form>
        <div className="form">
          <NameInput />
          <StreetInput />
          <div className="fieldRow">
            <CityInput />
            <StateInput />
            <ZipInput />
          </div>
          <div className="btn-row">
            <button type="submit" className="submit-btn">
              Submit
            </button>
          </div>
        </div>
      </Form>
    </Formik>
  );
};

export default AddressForm;
