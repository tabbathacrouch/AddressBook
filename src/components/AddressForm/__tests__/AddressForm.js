import React from "react";
import renderer from "react-test-renderer";
import AddressForm from "../AddressForm";

describe("<AddressForm />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<AddressForm />);
    expect(tree).toMatchSnapshot();
  });
});
