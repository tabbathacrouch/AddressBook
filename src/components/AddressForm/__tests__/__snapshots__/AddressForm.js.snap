// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressForm /> renders correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="form"
  >
    <div>
      <input
        className="field"
        data-testid="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="field"
        name="street"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Street"
        type="text"
        value=""
      />
    </div>
    <div
      className="fieldRow"
    >
      <div
        className="city"
      >
        <input
          className="field"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="City"
          type="text"
          value=""
        />
      </div>
      <div
        className="stateName"
      >
        <input
          className="field"
          name="stateName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="State"
          type="text"
          value=""
        />
      </div>
      <div
        className="zip"
      >
        <input
          className="field"
          name="zip"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Zip"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="btn-row"
    >
      <button
        className="submit-btn"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
